File: examples/hello-world/gabriel-greeting-cpp/internal/greetings.cpp
#include "internal/greetings.hpp"

namespace gabriel::greeting::cppholon::internal {
namespace {

constexpr std::array<Greeting, 56> kGreetings{{
    {"en", "English", "English", "Hello %s", "Mary"},
    {"fr", "French", "Français", "Bonjour %s", "Marie"},
    {"es", "Spanish", "Español", "Hola %s", "María"},
    {"de", "German", "Deutsch", "Hallo %s", "Maria"},
    {"it", "Italian", "Italiano", "Ciao %s", "Maria"},
    {"pt", "Portuguese", "Português", "Olá %s", "Maria"},
    {"nl", "Dutch", "Nederlands", "Hallo %s", "Maria"},
    {"ru", "Russian", "Русский", "Привет %s", "Мария"},
    {"ja", "Japanese", "日本語", "こんにちは、%sさん", "マリア"},
    {"zh", "Chinese", "中文", "你好，%s", "玛丽"},
    {"ko", "Korean", "한국어", "안녕하세요 %s", "마리아"},
    {"ar", "Arabic", "العربية", "مرحبا، %s", "ماريا"},
    {"hi", "Hindi", "हिन्दी", "नमस्ते %s", "मारिया"},
    {"tr", "Turkish", "Türkçe", "Merhaba %s", "Meryem"},
    {"pl", "Polish", "Polski", "Cześć %s", "Maria"},
    {"sv", "Swedish", "Svenska", "Hej %s", "Maria"},
    {"no", "Norwegian", "Norsk", "Hei %s", "Maria"},
    {"da", "Danish", "Dansk", "Hej %s", "Maria"},
    {"fi", "Finnish", "Suomi", "Hei %s", "Maria"},
    {"cs", "Czech", "Čeština", "Ahoj %s", "Marie"},
    {"ro", "Romanian", "Română", "Bună %s", "Maria"},
    {"hu", "Hungarian", "Magyar", "Szia %s", "Mária"},
    {"el", "Greek", "Ελληνικά", "Γεια σου %s", "Μαρία"},
    {"th", "Thai", "ไทย", "สวัสดี %s", "มาเรีย"},
    {"vi", "Vietnamese", "Tiếng Việt", "Xin chào %s", "Mary"},
    {"id", "Indonesian", "Bahasa Indonesia", "Halo %s", "Maria"},
    {"ms", "Malay", "Bahasa Melayu", "Hai %s", "Maria"},
    {"sw", "Swahili", "Kiswahili", "Habari %s", "Maria"},
    {"he", "Hebrew", "עברית", "שלום %s", "מרים"},
    {"uk", "Ukrainian", "Українська", "Привіт %s", "Марія"},
    {"bn", "Bengali", "বাংলা", "নমস্কার %s", "মারিয়া"},
    {"ta", "Tamil", "தமிழ்", "வணக்கம் %s", "மரியா"},
    {"fa", "Persian", "فارسی", "سلام، %s", "ماریا"},
    {"ur", "Urdu", "اردو", "السلام علیکم، %s", "ماریہ"},
    {"fil", "Filipino", "Filipino", "Kumusta %s", "Maria"},
    {"ca", "Catalan", "Català", "Hola %s", "Maria"},
    {"eu", "Basque", "Euskara", "Kaixo %s", "Maria"},
    {"gl", "Galician", "Galego", "Ola %s", "María"},
    {"is", "Icelandic", "Íslenska", "Halló %s", "María"},
    {"et", "Estonian", "Eesti", "Tere %s", "Maria"},
    {"lv", "Latvian", "Latviešu", "Sveiki %s", "Marija"},
    {"lt", "Lithuanian", "Lietuvių", "Sveiki %s", "Marija"},
    {"sk", "Slovak", "Slovenčina", "Ahoj %s", "Mária"},
    {"sl", "Slovenian", "Slovenščina", "Živjo %s", "Marija"},
    {"hr", "Croatian", "Hrvatski", "Bok %s", "Marija"},
    {"sr", "Serbian", "Српски", "Здраво %s", "Марија"},
    {"bg", "Bulgarian", "Български", "Здравей %s", "Мария"},
    {"ka", "Georgian", "ქართული", "გამარჯობა %s", "მარიამ"},
    {"hy", "Armenian", "Հայերեն", "Բարև %s", "Մարիա"},
    {"am", "Amharic", "አማርኛ", "ሰላም %s", "ማሪያ"},
    {"mn", "Mongolian", "Монгол", "Сайн уу %s", "Мария"},
    {"ne", "Nepali", "नेपाली", "नमस्कार %s", "मारिया"},
    {"kk", "Kazakh", "Қазақша", "Сәлем %s", "Мария"},
    {"uz", "Uzbek", "Oʻzbekcha", "Salom %s", "Mariya"},
    {"yo", "Yoruba", "Yorùbá", "Báwo %s", "Maria"},
    {"zu", "Zulu", "isiZulu", "Sawubona %s", "uMaria"},
}};

} // namespace

const std::array<Greeting, 56> &Greetings() { return kGreetings; }

const Greeting &Lookup(std::string_view lang_code) {
  for (const auto &entry : kGreetings) {
    if (entry.lang_code == lang_code) {
      return entry;
    }
  }
  return kGreetings.front();
}

} // namespace gabriel::greeting::cppholon::internal

File: examples/hello-world/gabriel-greeting-ruby/internal/greetings.rb
# frozen_string_literal: true

module GabrielGreetingRuby
  module Internal
    GreetingEntry = Struct.new(
      :lang_code,
      :lang_english,
      :lang_native,
      :template,
      :default_name,
      keyword_init: true
    )

    GREETINGS = [
      GreetingEntry.new(lang_code: "en", lang_english: "English", lang_native: "English", template: "Hello %s", default_name: "Mary"),
      GreetingEntry.new(lang_code: "fr", lang_english: "French", lang_native: "Français", template: "Bonjour %s", default_name: "Marie"),
      GreetingEntry.new(lang_code: "es", lang_english: "Spanish", lang_native: "Español", template: "Hola %s", default_name: "María"),
      GreetingEntry.new(lang_code: "de", lang_english: "German", lang_native: "Deutsch", template: "Hallo %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "it", lang_english: "Italian", lang_native: "Italiano", template: "Ciao %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "pt", lang_english: "Portuguese", lang_native: "Português", template: "Olá %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "nl", lang_english: "Dutch", lang_native: "Nederlands", template: "Hallo %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "ru", lang_english: "Russian", lang_native: "Русский", template: "Привет %s", default_name: "Мария"),
      GreetingEntry.new(lang_code: "ja", lang_english: "Japanese", lang_native: "日本語", template: "こんにちは、%sさん", default_name: "マリア"),
      GreetingEntry.new(lang_code: "zh", lang_english: "Chinese", lang_native: "中文", template: "你好，%s", default_name: "玛丽"),
      GreetingEntry.new(lang_code: "ko", lang_english: "Korean", lang_native: "한국어", template: "안녕하세요 %s", default_name: "마리아"),
      GreetingEntry.new(lang_code: "ar", lang_english: "Arabic", lang_native: "العربية", template: "مرحبا، %s", default_name: "ماريا"),
      GreetingEntry.new(lang_code: "hi", lang_english: "Hindi", lang_native: "हिन्दी", template: "नमस्ते %s", default_name: "मारिया"),
      GreetingEntry.new(lang_code: "tr", lang_english: "Turkish", lang_native: "Türkçe", template: "Merhaba %s", default_name: "Meryem"),
      GreetingEntry.new(lang_code: "pl", lang_english: "Polish", lang_native: "Polski", template: "Cześć %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "sv", lang_english: "Swedish", lang_native: "Svenska", template: "Hej %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "no", lang_english: "Norwegian", lang_native: "Norsk", template: "Hei %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "da", lang_english: "Danish", lang_native: "Dansk", template: "Hej %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "fi", lang_english: "Finnish", lang_native: "Suomi", template: "Hei %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "cs", lang_english: "Czech", lang_native: "Čeština", template: "Ahoj %s", default_name: "Marie"),
      GreetingEntry.new(lang_code: "ro", lang_english: "Romanian", lang_native: "Română", template: "Bună %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "hu", lang_english: "Hungarian", lang_native: "Magyar", template: "Szia %s", default_name: "Mária"),
      GreetingEntry.new(lang_code: "el", lang_english: "Greek", lang_native: "Ελληνικά", template: "Γεια σου %s", default_name: "Μαρία"),
      GreetingEntry.new(lang_code: "th", lang_english: "Thai", lang_native: "ไทย", template: "สวัสดี %s", default_name: "มาเรีย"),
      GreetingEntry.new(lang_code: "vi", lang_english: "Vietnamese", lang_native: "Tiếng Việt", template: "Xin chào %s", default_name: "Mary"),
      GreetingEntry.new(lang_code: "id", lang_english: "Indonesian", lang_native: "Bahasa Indonesia", template: "Halo %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "ms", lang_english: "Malay", lang_native: "Bahasa Melayu", template: "Hai %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "sw", lang_english: "Swahili", lang_native: "Kiswahili", template: "Habari %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "he", lang_english: "Hebrew", lang_native: "עברית", template: "שלום %s", default_name: "מרים"),
      GreetingEntry.new(lang_code: "uk", lang_english: "Ukrainian", lang_native: "Українська", template: "Привіт %s", default_name: "Марія"),
      GreetingEntry.new(lang_code: "bn", lang_english: "Bengali", lang_native: "বাংলা", template: "নমস্কার %s", default_name: "মারিয়া"),
      GreetingEntry.new(lang_code: "ta", lang_english: "Tamil", lang_native: "தமிழ்", template: "வணக்கம் %s", default_name: "மரியா"),
      GreetingEntry.new(lang_code: "fa", lang_english: "Persian", lang_native: "فارسی", template: "سلام، %s", default_name: "ماریا"),
      GreetingEntry.new(lang_code: "ur", lang_english: "Urdu", lang_native: "اردو", template: "السلام علیکم، %s", default_name: "ماریہ"),
      GreetingEntry.new(lang_code: "fil", lang_english: "Filipino", lang_native: "Filipino", template: "Kumusta %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "ca", lang_english: "Catalan", lang_native: "Català", template: "Hola %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "eu", lang_english: "Basque", lang_native: "Euskara", template: "Kaixo %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "gl", lang_english: "Galician", lang_native: "Galego", template: "Ola %s", default_name: "María"),
      GreetingEntry.new(lang_code: "is", lang_english: "Icelandic", lang_native: "Íslenska", template: "Halló %s", default_name: "María"),
      GreetingEntry.new(lang_code: "et", lang_english: "Estonian", lang_native: "Eesti", template: "Tere %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "lv", lang_english: "Latvian", lang_native: "Latviešu", template: "Sveiki %s", default_name: "Marija"),
      GreetingEntry.new(lang_code: "lt", lang_english: "Lithuanian", lang_native: "Lietuvių", template: "Sveiki %s", default_name: "Marija"),
      GreetingEntry.new(lang_code: "sk", lang_english: "Slovak", lang_native: "Slovenčina", template: "Ahoj %s", default_name: "Mária"),
      GreetingEntry.new(lang_code: "sl", lang_english: "Slovenian", lang_native: "Slovenščina", template: "Živjo %s", default_name: "Marija"),
      GreetingEntry.new(lang_code: "hr", lang_english: "Croatian", lang_native: "Hrvatski", template: "Bok %s", default_name: "Marija"),
      GreetingEntry.new(lang_code: "sr", lang_english: "Serbian", lang_native: "Српски", template: "Здраво %s", default_name: "Марија"),
      GreetingEntry.new(lang_code: "bg", lang_english: "Bulgarian", lang_native: "Български", template: "Здравей %s", default_name: "Мария"),
      GreetingEntry.new(lang_code: "ka", lang_english: "Georgian", lang_native: "ქართული", template: "გამარჯობა %s", default_name: "მარიამ"),
      GreetingEntry.new(lang_code: "hy", lang_english: "Armenian", lang_native: "Հայերեն", template: "Բարև %s", default_name: "Մարիա"),
      GreetingEntry.new(lang_code: "am", lang_english: "Amharic", lang_native: "አማርኛ", template: "ሰላም %s", default_name: "ማሪያ"),
      GreetingEntry.new(lang_code: "mn", lang_english: "Mongolian", lang_native: "Монгол", template: "Сайн уу %s", default_name: "Мария"),
      GreetingEntry.new(lang_code: "ne", lang_english: "Nepali", lang_native: "नेपाली", template: "नमस्कार %s", default_name: "मारिया"),
      GreetingEntry.new(lang_code: "kk", lang_english: "Kazakh", lang_native: "Қазақша", template: "Сәлем %s", default_name: "Мария"),
      GreetingEntry.new(lang_code: "uz", lang_english: "Uzbek", lang_native: "Oʻzbekcha", template: "Salom %s", default_name: "Mariya"),
      GreetingEntry.new(lang_code: "yo", lang_english: "Yoruba", lang_native: "Yorùbá", template: "Báwo %s", default_name: "Maria"),
      GreetingEntry.new(lang_code: "zu", lang_english: "Zulu", lang_native: "isiZulu", template: "Sawubona %s", default_name: "uMaria")
    ].freeze

    INDEX = GREETINGS.each_with_object({}) do |entry, index|
      index[entry.lang_code] = entry
    end.freeze

    class << self
      def lookup(code)
        INDEX.fetch(code.to_s, INDEX.fetch("en"))
      end
    end
  end
end
